mars-cli/mars_lib/authentication.py

```diff
@@ -1,6 +1,66 @@
-from typing import Optional
+import io
+from typing import Optional, Union
 import requests
 import json
+from enum import Enum
+
+
+class AuthProvider(Enum):
+    """
+    Holds constants, tied to the repository authentication providers.
+    """
+
+    WEBIN = "webin"
+    METABOLIGHTS_METADATA = "metabolights_metadata"
+    METABOLIGHTS_DATA = "metabolights_data"
+
+    @classmethod
+    def available_providers(cls):
+        return {item.value for item in cls}
+
+    @classmethod
+    def is_valid_provider(cls, provider: str):
+        return provider in cls.available_providers()
+
+
+def load_credentials(
+    credentials_file: Union[io.TextIOWrapper, str]
+) -> dict[str, dict[str, str]]:
+    """
+    Validate the credentials.
+
+    Args:
+    credentials_file (_): The credentials in file formate.
+
+    Raises:
+    ValueError: If the credentials are not valid.
+
+    Returns:
+    dict: The credentials.
+    """
+    if isinstance(credentials_file, str):
+        with open(credentials_file, "r") as file:
+            credentials = json.load(file)
+    elif isinstance(credentials_file, io.TextIOWrapper):
+        with open(credentials_file.name, "r") as file:
+            credentials = json.load(file)
+    else:
+        raise TypeError("Credentials file must be of type str or io.TextIOWrapper.")
+
+    if not all(
+        repo in AuthProvider.available_providers() for repo in credentials.keys()
+    ):
```

Comment:
I think eventually should loosen this up, e.g. if I'm only submitting the ENA, EVA and BioSamples I shouldn't need Metabolights credentials... it's fine for now though.

Reply:
I agree!